src/heretic/model.py

```diff
@@ -57,8 +62,14 @@ def __init__(self, settings: Settings):
                     settings.model,
                     dtype=dtype,
                     device_map=settings.device_map,
+                    trust_remote_code=self.trusted_models.get(settings.model),
                 )
 
+                # If we reach this point and the model requires trust_remote_code,
+                # the user must have confirmed it.
+                if self.trusted_models.get(settings.model) is None:
+                    self.trusted_models[settings.model] = True
```

Comment:
:exclamation: :white_check_mark: This is the critical part, and the comment correctly describes why it works. Declining the prompt for a model requiring `trust_remote_code` will raise an exception, so at this point, we know that the user has either confirmed the prompt or passed `--trust-remote-code True` to begin with.